packages/analytics-connector/src/eventBridge.ts

```diff
@@ -9,16 +14,58 @@ export type AnalyticsEventReceiver = (event: AnalyticsEvent) => void;
 export interface EventBridge {
   logEvent(event: AnalyticsEvent): void;
   setEventReceiver(listener: AnalyticsEventReceiver): void;
+  setApiKey(apiKey: string): void;
 }
 
 export class EventBridgeImpl implements EventBridge {
+  private apiKey = '';
   private receiver: AnalyticsEventReceiver;
-  private queue: AnalyticsEvent[] = [];
+  private inMemoryQueue: AnalyticsEvent[] = [];
+  private globalScope = getGlobalScope();
+
+  private getStorageKey(): string {
+    return `EXP_unsent_${this.apiKey.slice(0, 10)}`;
+  }
+
+  private initializeStorage(): void {
+    if (isLocalStorageAvailable()) {
+      const storageKey = this.getStorageKey();
+      const storedQueue = this.globalScope.localStorage.getItem(storageKey);
+      if (storedQueue === null) {
+        // Initialize with an empty array if no queue is found
+        this.globalScope.localStorage.setItem(storageKey, JSON.stringify([]));
+      }
+    }
+  }
+
+  private getQueue(): AnalyticsEvent[] {
+    if (isLocalStorageAvailable()) {
+      const storageKey = this.getStorageKey();
+      const storedQueue = this.globalScope.localStorage.getItem(storageKey);
+      return storedQueue ? JSON.parse(storedQueue) : [];
+    } else {
+      return this.inMemoryQueue;
+    }
+  }
+
+  private setQueue(queue: AnalyticsEvent[]): void {
+    if (isLocalStorageAvailable()) {
+      this.globalScope.localStorage.setItem(
+        this.getStorageKey(),
+        JSON.stringify(queue),
+      );
+    } else {
+      this.inMemoryQueue = queue;
+    }
+  }
 
   logEvent(event: AnalyticsEvent): void {
     if (!this.receiver) {
-      if (this.queue.length < 512) {
-        this.queue.push(event);
+      this.initializeStorage(); // Ensure storage is initialized
```

Comment:
Calling this we are always calling getItem on local storage. Once we initialize the storage, we should store that info in memory in a boolean and only initialize the storage if we have not already in this session.